deploy/eck-agent/values.yaml

```diff
@@ -0,0 +1,165 @@
+---
+# Default values for eck-agent.
+# This is a YAML-formatted file.
+
+# Overridable names of the Elastic Agent resource.
+# By default, this is the Release name set for the chart,
+# followed by 'eck-agent'.
+#
+# nameOverride will override the name of the Chart with the name set here,
+# so nameOverride: quickstart, would convert to '{{ Release.name }}-quickstart'
+#
+# nameOverride: "quickstart"
+#
+# fullnameOverride will override both the release name, and the chart name,
+# and will name the Fleet Agent resource exactly as specified.
+#
+# fullnameOverride: "quickstart"
+
+# Version of Elastic Agent.
+#
+version: 8.2.3
+
+# Labels that will be applied to Elastic Agent.
+#
+labels: {}
+
+# Annotations that will be applied to Elastic Agent.
+#
+annotations: {}
+
+spec:
+  # Referenced resources are below and depending on the setup, at least one is required for a functional Agent.
+  # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-elastic-agent-fleet-configuration.html#k8s-elastic-agent-fleet-configuration-setting-referenced-resources
+  #
+  # Reference to ECK-managed Kibana instance.
+  #
+  # kibanaRef:
+  #   name: quickstart
+    # Optional namespace reference to Kibana instance.
+    # If not specified, then the namespace of the Agent instance
+    # will be assumed.
+    #
+    # namespace: default
+
+  # Reference to ECK-managed Elasticsearch instance.
+  #
+  elasticsearchRefs:
+  - name: elasticsearch
+    # Optional namespace reference to Elasticsearch instance.
+    # If not specified, then the namespace of the Agent instance
+    # will be assumed.
+    #
+    # namespace: default
+  
+  # Reference to ECK-managed Fleet Server instance.
+  #
+  # fleetServerRef:
+  #   name: fleet-server
+    # Optional namespace reference to Fleet Server instance.
+    # If not specified, then the namespace of the Agent instance
+    # will be assumed.
+    #
+    # namespace: default
+
+  # Daemonset, or Deployment specification for Agent.
+  # At least one is required of [daemonSet, deployment].
+  # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-elastic-agent-fleet-configuration.html#k8s-elastic-agent-fleet-configuration-chose-the-deployment-model
+  #
+  daemonSet:
+    podTemplate:
+      spec:
+        containers:
+        - name: agent
+          securityContext:
+            runAsUser: 0
+
+  # Configuration of Agent, specifically used in Agent standalone mode.
+  # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-elastic-agent.html
+  #
+  config: {}
```

Comment:
This will lead to errors when running in Fleet mode but due to the Helm issues it cannot be nullified. We could relax the validation in the operator to tread `config: nil` and `config: {}` the same.

Reply:
> This will lead to errors when running in Fleet mode but due to the Helm issues it cannot be nullified via a values override. We could relax the validation in the operator to tread `config: nil` and `config: {}` the same.

I went ahead and tested this, as I thought I had run though this process a couple weeks ago, and I can't replicate this issue.  The validating webhook allows this; all pods come online; I don't see any errors in fleet-server, or agent pod logs.  Can you expound on where the errors from this might be @pebrc ?

```
❯ helm install eck-stack . -n default --values ./examples/fleet-agents.yaml
NAME: eck-stack
LAST DEPLOYED: Tue Sep 20 12:15:01 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Elasticsearch ECK-Stack 0.1.0 has been deployed successfully!

To see status of all resources, run

kubectl get elastic -n default -l "app.kubernetes.io/instance"=eck-stack

More information on the Elastic ECK Operator, and its Helm chart can be found
within our documentation.

https://www.elastic.co/guide/en/cloud-on-k8s/current/index.html

❯ kc get MutatingWebhookConfiguration,ValidatingWebhookConfiguration -A | grep elastic
validatingwebhookconfiguration.admissionregistration.k8s.io/elastic-operator.elastic-system.k8s.elastic.co                    10         21m

❯ kc get pods -n default
NAME                                  READY   STATUS    RESTARTS   AGE
eck-stack-eck-agent-agent-9nhhf       1/1     Running   0          2m30s
eck-stack-eck-agent-agent-gvq4n       1/1     Running   0          2m30s
eck-stack-eck-agent-agent-s8p2k       1/1     Running   0          2m30s
elasticsearch-es-default-0            1/1     Running   0          8m8s
elasticsearch-es-default-1            1/1     Running   0          8m8s
elasticsearch-es-default-2            1/1     Running   0          8m8s
fleet-server-agent-6fddd99dcd-h8679   1/1     Running   0          2m31s
kibana-kb-6766b97bf7-k6z8w            1/1     Running   0          8m4s
```

Reply:
Just tried once more (I am using a fresh 2.4.0 install, yesterday I was using the operator code from your branch) 

```
helm install eck-stack . -n default --values ./examples/fleet-agents.yaml
Error: INSTALLATION FAILED: admission webhook "elastic-agent-validation-v1alpha1.k8s.elastic.co" denied the request: Agent.agent.k8s.elastic.co "eck-stack-eck-agent" is invalid: spec.config: Invalid value: v1.Config{Data:map[string]interface {}{}}: remove config, it can't be set in fleet mode
```

Reply:
@pebrc Agent default value for `spec.config` moved to `null` instead of `{}` to resolve.  Let me know if you see any further issues.